### PR TITLE
Correction for my issue

### LINKED ISSUE #3: Captcha error в APIError
У меня есть одна просьба, можете ли вы вывести результат ошибки через функцию to_json в тип Value? Или же как то выдать ссылку на саму каптчу? Могу предложить Pull Request если нужно указать поправку

### PR DESCRIPTION
**Pull request by TheDeaX-Lab**
Fixes #3

### CODE REVIEW

File: src/error.rs [hunk 1]
```diff
@@ -7,19 +7,21 @@ use serde_derive::Deserialize;
 pub type Result<T> = std::result::Result<T, Error>;
 
 /// An error returned by the API
-#[derive(Fail, Deserialize, Debug, Eq, PartialEq, Clone)]
+#[derive(Fail, Deserialize, Debug, PartialEq, Eq, Clone)]
 #[fail(display = "API Error #{}: {}", error_code, error_msg)]
 pub struct APIError {
     error_code: u64,
     error_msg: String,
+    error_value_str: String,
```
Comment (TheDeaX-Lab) [inline src/error.rs]: Adding new param for error

**Comment by awaitlink** [comment]
Thanks for your contribution, but I think there is an easier way to do what you're trying to do.

As far as I understand, you want to catch all extra fields in the error object.

We can use [`#[serde(flatten)]`](https://serde.rs/attr-flatten.html#capture-additional-fields) here:
```rust
struct APIError {
    // ...

    #[serde(flatten)]
    extra: HashMap<String, Value>,
}
```

Then there is no need for modifying the error creation logic in `APIClient` and adding stuff like `to_json`. Only in `APIError::new` we will need to accept `extra: HashMap<String, Value>`. You can probably also derive `Serialize` for `APIError`, and then you will be able to serialize `APIError` back into `Value`.

I will be happy to accept your PR once you make the changes :)

**Comment by TerryChan2003** [comment]
> Thanks for your contribution, but I think there is an easier way to do what you're trying to do.
> 
> As far as I understand, you want to catch all extra fields in the error object.
> 
> We can use [`#[serde(flatten)]`](https://serde.rs/attr-flatten.html#capture-additional-fields) here:
> 
> ```rust
> struct APIError {
>     // ...
> 
>     #[serde(flatten)]
>     extra: HashMap<String, Value>,
> }
> ```
> 
> Then there is no need for modifying the error creation logic in `APIClient` and adding stuff like `to_json`. Only in `APIError::new` we will need to accept `extra: HashMap<String, Value>`. You can probably also derive `Serialize` for `APIError`, and then you will be able to serialize `APIError` back into `Value`.
> 
> I will be happy to accept your PR once you make the changes :)
i have error
```The attribute `serge` is currently unknown to the compiler and may have meaning added to it in the future (see issue #29642)```

**Comment by awaitlink** [comment]
`serge` != `serde` :)

**Comment by TheDeaX-Lab** [comment]
not works only

The attribute `serde` is currently unknown to the compiler and may have meaning added to it in the future (see issue #29642)

Do you speak in russian? :D 

The solution is great, but I'm still not so strong on this one.

**Comment by awaitlink** [comment]
Strange, for me it works, do you have latest stable rustc?

Also, we need to remove derive `Eq` for this to work.

**Comment by TheDeaX-Lab** [comment]
Of course, the latest version that can be for ArchLinux.

**Comment by TheDeaX-Lab** [comment]
https://ibb.co/GJhy9mf rust version 1.32

**Comment by awaitlink** [comment]
Ah, I see, you are putting the attribute in the wrong place, see the [page I linked to earlier](https://serde.rs/attr-flatten.html#capture-additional-fields). You need to add it on the struct field, not on its initialization.

**Comment by TheDeaX-Lab** [comment]
oh sorry not threw there, so ... Now the last mistake https://ibb.co/6vtJ59J .

**Comment by awaitlink** [comment]
Yep, remove derive `Eq`, as I said earlier. I guess there is a good reason why `Value`s are not `Eq`.

**Comment by TheDeaX-Lab** [comment]
and another mistake https://ibb.co/kHpHLf6

**Comment by awaitlink** [comment]
Please read what I replied to you initially carefully :)

> Only in `APIError::new` we will need to accept `extra: HashMap<String, Value>`

**Comment by TheDeaX-Lab** [comment]
and again error yes what again then https://ibb.co/nfkMbKv

**Comment by awaitlink** [comment]
In those tests, do something like this:
```rust
use rvk::error::APIError;
use std::collections::HashMap;

let err = APIError::new(0, "test".to_string(), HashMap::new());
// ...
```

**Comment by TheDeaX-Lab** [comment]
Now the errors are fixed. I prepare pull request

File: src/error.rs [hunk 2]
```diff
@@ -1,34 +1,40 @@
...
 }
 
 impl APIError {
     /// Creates a new `APIError`
-    pub fn new(code: u64, msg: String) -> Self {
+    pub fn new(code: u64, msg: String, extra: HashMap<String, Value>) -> Self {
         Self {
             error_code: code,
             error_msg: msg,
+            extra: extra,
```
Comment (awaitlink) [inline src/error.rs]: You can use the field init shorthand here:
```suggestion
            extra,
```

**Review by awaitlink — APPROVED** [review]
LGTM, waiting for Travis

**Pull request merged**